src/app/src/components/UtilityGuard.js

```diff
@@ -25,6 +27,25 @@ export default function UtilityGuard({ children }) {
 
 function ContributorUtilityGuard({ children }) {
     const dispatch = useDispatch();
+    const { pathname } = useLocation();
+
+    let boundaryId = null;
+
+    if (pathname.startsWith('/submissions/')) {
+        boundaryId = pathname.substring('/submissions/'.length);
+        if (pathname.endsWith('/draw')) {
+            boundaryId = boundaryId.substring(
+                0,
+                boundaryId.length - '/draw'.length
+            );
+        }
+    }
```

Comment:
```suggestion
    const boundaryId = pathname.match(/^\/submissions\/(\d+)/)?.[1];
```

Reply:
Perfect. Fixed in [b555cbc](https://github.com/azavea/iow-boundary-tool/pull/218/commits/b555cbc95cb4f6234bde07de95fc5e8bf577a613)